### PR TITLE
add docstrings and typing in zos SAL

### PR DESCRIPTION
**Pull request by zaibon**

### CODE REVIEW

**Comment by waleedhammam** [comment]
Why do we change the docstrings convention, it doesn't follow with js-ng ?

File: jumpscale/sals/zos/container.py [hunk 1]
```diff
@@ -1,60 +1,71 @@
...
-            env (dict, optional): Environment variables to set. Defaults to {}.
-            cpu (int, optional): CPU capacity. Defaults to 1.
-            memory (int, optional): Memory capacity. Defaults to 1024.
-            entrypoint (str, optional): Container init command. Defaults to "".
-            interactive (bool, optional): Specifies interactive contaienr start or not. Defaults to False.
-            secret_env (dict, optional): Secret Environment variables to set. Defaults to {}.
-            public_ipv6 (bool, optional): IPV6 container ip address in the network. Defaults to False.
-            storage_url (str, optional): Server used as storage backend. Defaults to "zdb://hub.grid.tf:9900".
-
-        Raises:
```
Comment (xmonader) [inline jumpscale/sals/zos/container.py:53]: here we lost the expected exceptions is there a way to include in the docstrings?

File: jumpscale/sals/zos/kubernetes.py [hunk 2]
```diff
@@ -1,30 +1,44 @@
...
-            reservation (jumpscale.clients.explorer.models.TfgridWorkloadsReservation1): reservation object
-            node_id (str): Id of the node to be added
-            network_name (str): name of the network to join
-            cluster_secret (str): Secret of the cluster to be joined
-            ip_address (str): Ip address of the master node
-            size (int): size of master node, either 1 or 2
-            ssh_keys (list, optional): List of ssh keys to be added to that node. Defaults to [].
-
-        Raises:
-            jumpscale.core.exceptions.Input: If size is not supported
```
Comment (xmonader) [inline jumpscale/sals/zos/kubernetes.py:24]: we lost the raised exceptions info here too

File: jumpscale/sals/zos/kubernetes.py [hunk 3]
```diff
@@ -45,24 +59,38 @@ def add_master(self, node_id, network_name, cluster_secret, ip_address, size, ss
...
-            reservation (jumpscale.clients.explorer.models.TfgridWorkloadsReservation1): reservation object
-            node_id (str): Id of the node to be added
-            network_name (str): name of the network to join
-            cluster_secret (str): Secret of the cluster to be joined
-            ip_address (str): Ip address of the worker node
-            size (int): size of master node, either 1 or 2
-            master_ip (str): IP address of a master node
-            ssh_keys (list, optional): List of ssh keys to be added to that node. Defaults to [].
-
-        Raises:
```
Comment (xmonader) [inline jumpscale/sals/zos/kubernetes.py:61]: same for losing exceptions

File: jumpscale/sals/zos/network.py [hunk 4]
```diff
@@ -42,20 +42,16 @@ def _cleanup_network(self, network):
...
 
-    def create(self, ip_range, network_name=None):
-        """add a network into the reservation
-
-        Args:
-            reservation (jumpscale.clients.explorer.models.TfgridWorkloadsReservation1]): root reservation object, the network will be added to it
-            ip_range (str): subnet of the network, it must have a network mask of /16
-            network_name (str, optional): identifier of the network, if not specified a randon name will be generated. Defaults to None.
-
-        Raises:
```
Comment (xmonader) [inline jumpscale/sals/zos/network.py:53]: we lost that section for exceptions as well

File: jumpscale/sals/zos/network.py [hunk 5]
```diff
@@ -106,25 +103,25 @@ def add_node(self, network, node_id, ip_range, pool_id, wg_port=None):
...
-           a member to your network that is not a 0-OS node like User laptop, external server,...
-
-        Args:
-            network ([type]): network object where to add the network resource
-            node_id (str): node_id of the node to use a access point into the network
-            ip_range (str): subnet to allocate to the member,  network mask should be a /24 and be part of the network subnet
-            wg_public_key (str, optional): public key of the new member. If none a new key pair will be generated automatically. Defaults to None.
-            ipv4 (bool, optional): if True, the endpoint of the access node will use IPv4. Use this if the member is not IPv6 enabled. Defaults to False.
-
-        Raises:
```
Comment (xmonader) [inline jumpscale/sals/zos/network.py:120]: lost exceptions too

File: jumpscale/sals/zos/zdb.py [hunk 6]
```diff
@@ -1,31 +1,44 @@
...
-        Args:
-            reservation (jumpscale.clients.explorer.models.TfgridWorkloadsReservation1): reservation obejct to add the volume to
-            node_id (str): id of the node to reserve the volume
-            size (int): size of the namespace in GiB
-            mode (str): mode of the 0-db, can be 'seq' or 'user'
-            password (str): password of the namespace. if you don't want password use an empty string
-            disk_type (str, optional): type of disk,can be SSD or HDD. Defaults to "SSD".
-            public (bool, optional): if public is True, anyone can write to the namespace without being authenticated. Defaults to False.
-
-        Raises:
```
Comment (xmonader) [inline jumpscale/sals/zos/zdb.py:21]: we lost exceptions here again

**Comment by zaibon** [comment]
> Why do we change the docstrings convention, it doesn't follow with js-ng ? 

as far as I know the one I use is the convention

**Comment by zaibon** [comment]
ha.... I just saw that the convention changes between jsx and js-ng....
So now we use google-stype docstring ??

File: jumpscale/sals/zos/container.py (hunk 1 above)
Comment (zaibon) [reply jumpscale/sals/zos/container.py:53]: it's because the function doesn't raise this exception anymore.

**Comment by zaibon** [comment]
closed in favor of https://github.com/threefoldtech/js-sdk/pull/413

**Pull request closed without merging**